src/tiled/abstractobjecttool.cpp

```diff
@@ -246,6 +247,74 @@ void AbstractObjectTool::removeObjects()
     mapDocument()->removeObjects(mapDocument()->selectedObjects());
 }
 
+/**
+ * @brief AbstractObjectTool::applyCollisionsToSelection
+ *
+ * Add the selected collision masks for the currently selected tile - the last selected tile - to
+ * all selected tiles.
+ */
+void AbstractObjectTool::applyCollisionsToSelection()
+{
+    // The selected Collision Masks
+    QList<MapObject*> selectedObjects = mapDocument()->selectedObjects();
+
+    if (auto document = DocumentManager::instance()->currentDocument()) {
+        if (auto tilesetDocument = qobject_cast<TilesetDocument*>(document)) {
+            if (Tile *currentTile = dynamic_cast<Tile *>(tilesetDocument->currentObject())) {
+
+                // Ask the user if they want to append or overwrite the collision mask
+                QMessageBox msgBox;
+                msgBox.setIcon(QMessageBox::Information);
+                msgBox.setWindowTitle(tr("Overwrite Collision Masks?"));
+                msgBox.setText(tr("Should Collision Masks be appended to, or replace, selected tiles' masks?"));
+                msgBox.addButton(tr("Append"), QMessageBox::ActionRole);
+                QPushButton *overwriteBtn = msgBox.addButton(tr("Replace"), QMessageBox::ActionRole);
+                QPushButton *abortButton = msgBox.addButton(QMessageBox::Cancel);
+                msgBox.setDefaultButton(overwriteBtn);
+                msgBox.exec();
+
+                if (msgBox.clickedButton() != abortButton) {
+
+                    QList<Tile *>selectedTiles = tilesetDocument->selectedTiles();
```

Comment:
Since we're not going to change the list of selected tiles, this can be:
```c++
const auto &selectedTiles = tilesetDocument->selectedTiles();
```
Or even just:
```c++
for (Tile *tile : tilesetDocument->selectedTiles()) {
```